engines/producers/spec/controllers/producers/api/v1/products_controller_spec.rb

```diff
@@ -0,0 +1,429 @@
+              let(:params) do
+                {
+                  name: 'Tomatekken',
+                  price: 10.13,
+                  unit: Product::UNITS[:kg],
+                  producer_id: producer_for_user.id
+                }
+              end
+
+              it_behaves_like 'a successful request'
+
+              describe 'its body' do
+                before { post :create, params }
+
+                subject { JSON.parse(response.body) }
+
+                its(['name']) { is_expected.to eq(params[:name]) }
+                its(['price']) { is_expected.to eq(params[:price].to_s) }
+                its(['unit']) { is_expected.to eq(params[:unit]) }
+                its(['producer_id']) { is_expected.to eq(params[:producer_id]) }
+              end
+
+              context 'and the user is not a producer admin' do
+                before do
+                  producer_membership = Membership.where(
+                    user: producers_user,
+                    producer: producer_for_user
+                  ).first
+                  producer_membership.role = Membership::ROLES[:member]
+                  producer_membership.save
+                end
+
+                it_behaves_like 'a forbidden request'
+              end
+            end
+
+            context 'when the producer is associated to the user through a group' do
+              let(:params) do
+                {
+                  name: 'Tomatekken',
+                  price: 10.13,
+                  unit: Product::UNITS[:kg],
+                  producer_id: producer_for_group.id
+                }
+              end
+
+              it_behaves_like 'a successful request'
+
+              describe 'its body' do
+                before { post :create, params }
+
+                subject { JSON.parse(response.body) }
+
+                its(['name']) { is_expected.to eq(params[:name]) }
+                its(['price']) { is_expected.to eq(params[:price].to_s) }
+                its(['unit']) { is_expected.to eq(params[:unit]) }
+                its(['producer_id']) { is_expected.to eq(params[:producer_id]) }
+              end
+
+              context 'and the user is not a group admin' do
+                before do
+                  group_membership.role = ::Group::Membership::ROLES[:member]
+                  group_membership.save
+                end
+
+                it_behaves_like 'a forbidden request'
+              end
+            end
+
+            context 'with wrong parameters' do
+              let(:params) do
+                {
+                  name: 'Tomatekken',
+                  unit: 42,
+                  producer_id: producer_for_group.id
+                }
+              end
+
+              it_behaves_like 'a bad request'
+
+              describe 'its body' do
+                before { post :create, params }
+
+                subject { JSON.parse(response.body) }
+
+                its(:size) { is_expected.to eq(2) }
```

Comment:
Imo we can get rid of this expectation by simply checking that the response contains exactly the two elements below. You can use at least match_array or contaib_exactly matchers. This will save precious performance